src/common/evaluable.ts

```diff
@@ -46,6 +49,10 @@ export interface Evaluable {
    */
   evaluate(ctx: Context): Result;
 
+  simplify(ctx: Context): Result | Evaluable
```

Comment:
Missing trailing `;` -- not sure if that matters, but for consistency.

Reply:
I would say let's remove the ```;``` it may a legacy stuff